### DISCUSSION TITLE
Developing custom theme

### DISCUSSION THREAD

**Discussion by palexdev** [shannah/jdeploy#107] [Q&A]
@shannah Hello, so, today I'm trying to understand how to build a custom theme
I downloaded the fx-theme directory, but it seems like you never published the API:
![image](https://github.com/shannah/jdeploy/assets/16880178/771bc129-8cfb-4561-bbf3-9008ce97fa55)

**Comment by shannah** [comment]
Themes like this are a part of the jDeploy installer so it would need to be incorporated into jDeploy and I would need to make a new release.  This is necessary because I am signing the code of the installers and need to ensure that no unverified code is run as part of the installer.

So, you need to clone the jdeploy repo and do an `mvn install` on the installer directory to get that artifact locally.

Frankly, I'm not sure that developing an installer with JavaFX is the best way to go.  An installer is so simple, and Swing can achieve pretty much any UI/look needed.  (E.g. See The Swing Flat LaF, or IntelliJ IDEA).  

Maybe a better approach is to just create a mock up (just an image) of what you think the installer should look like.  I may be able to parametrize it to allow a more "css-ish-only" type theme.

**Comment by palexdev** [comment]
While projects like Swing Flat LaF are good indeed, I ensure you that Swing, because of its age, will never reach the levels of JavaFX, the things you can achieve with CSS are incredible. That said, I honestly don't quite understand why it is handled like that.
Instead of themes, can't we just build a custom installer?

I was thinking something like: I develop my UI (frontend) in whatever toolkit I want to use. Then my components would call the installer API (backend) to execute the usual actions (check for updates, unpackage,...)
